src/shared/Microsoft.Git.CredentialManager/EnvironmentVariables.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Microsoft.Git.CredentialManager
+{
+    /// <summary>
+    /// Simple wrapper dictionary around environment variables.
+    /// </summary>
+    public interface IEnvironmentVariables : IReadOnlyDictionary<string, string> { }
+
+    public class EnvironmentVariables : IEnvironmentVariables
+    {
+        private readonly IReadOnlyDictionary<string, string> _envars;
+
+        public EnvironmentVariables(IDictionary variables)
+        {
+            EnsureArgument.NotNull(variables, nameof(variables));
+
+            // On Windows it is technically possible to get env vars which differ only by case
+            // even though the general assumption is that they are case insensitive on Windows.
+            // For example, some of the standard .NET types like System.Diagnostics.Process
+            // will fail to start a process on Windows if given duplicate environment variables.
+            // See this issue for more information: https://github.com/dotnet/corefx/issues/13146
+
+            // If we're on the Windows platform we should de-duplicate by setting the string
+            // comparer to OrdinalIgnoreCase.
+            var comparer = PlatformUtils.IsWindows()
+                ? StringComparer.OrdinalIgnoreCase
+                : StringComparer.Ordinal;
```

Comment:
That is correct. This is the best we can do on Windows to avoid this situation, which only happens in obscure circumstances and only if invoked from Git (which can play fast-and-loose with the PEB).